src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3816,27 +3890,28 @@ private static void FixPartialProperty(ref Dictionary<ReadOnlyMemory<char>, Immu
             if (implementation.DeclaredBackingField is { } implementationField &&
                 definition.DeclaredBackingField is { })
             {
-                var fieldName = implementationField.Name.AsMemory();
-                membersByName[fieldName] = Remove(membersByName[fieldName], implementationField);
+                Remove(nonTypeMembers, implementationField);
             }
 
             SourcePropertySymbol.InitializePartialPropertyParts(definition, implementation);
 
             // a partial property is represented in the member list by its definition part:
-            membersByName[name] = Remove(membersByName[name], implementation);
+            Remove(nonTypeMembers, implementation);
         }
 
-        private static ImmutableArray<Symbol> Remove(ImmutableArray<Symbol> symbols, Symbol symbol)
+        private static void Remove(ArrayBuilder<Symbol> symbols, Symbol symbol)
```

Comment:
>ArrayBuilder<Symbol> symbols [](http://example.com/codeflow?start=35&length=28)

It looks like we call `Remove` once for each partial method or property that has both parts. Does the `symbols` array contain all non-type members? If that's the case, the number of `ReferenceEquals()` calls across all `Remove` calls for the containing type will be roughly N*M where N is the number of non-type members and M is the number of partial members. Are there cases where that might be an issue?

Reply:
> Are there cases where that might be an issue?

It is hard to tell for sure. I assume that ```ReferenceEquals``` call is pretty fast, since it is optimized away and probably roughly close to a pointer comparison. Also, M is likely not very big and no more than N/2. Also, on average, the match will be found before we check all N items, and each removal decreases N by one.

Reply:
The `M` that I am using is not the number of partial members. It is the number of partial members that we are going to remove, which is no more than the number of partial members divided by two.